frontend/src/components/Recipes/RecipesList.tsx

```diff
@@ -1,7 +1,29 @@
+import { useSearchParams } from 'react-router'
 import { RecipeCard } from '.'
-import { Recipe } from '../../api/recipe'
+import { endpoint, LIMIT, Recipe } from '../../api/recipe'
+import { useData } from '../../hooks/useData'
+
+export default function RecipesList() {
+    const [searchParams] = useSearchParams()
+    const page = Number(searchParams.get('_page')) || 1
+
+    const {
+        data: recipes = [],
+        error,
+        loading,
+    } = useData<Recipe[]>(
+        `${endpoint}/recipes?_page=${page}&_limit=${LIMIT}&_expand=difficulty`,
+        []
+    )
+
+    if (error) {
+        return <div>Error: {error.message}</div>
+    }
+
+    if (loading || recipes.length === 0) {
+        return <div>No recipes found</div>
+    }
```

Comment:
It is always better to distinguish the rendering of the loading state from a successful request with no data.
Here is may seem the same because the initial state is [] but you may be fetching and the user already sees that there are no recipes